pkg/operation/operation.go

```diff
@@ -502,30 +500,49 @@ func (o *Operation) InjectShootShootImages(values map[string]interface{}, names
 	return chart.InjectImages(values, o.ImageVector, names, imagevector.RuntimeVersion(o.ShootVersion()), imagevector.TargetVersion(o.ShootVersion()))
 }
 
-// EnsureShootStateExists creates the ShootState resource for the corresponding shoot and sets its ownerReferences to the Shoot.
+// EnsureShootStateExists creates the ShootState resource for the corresponding shoot and updates the operations object
 func (o *Operation) EnsureShootStateExists(ctx context.Context) error {
+	var (
+		err        error
+		shootState = &gardencorev1alpha1.ShootState{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      o.Shoot.Info.Name,
+				Namespace: o.Shoot.Info.Namespace,
+			},
+		}
+	)
+
+	if err = o.K8sGardenClient.Client().Create(ctx, shootState); err != nil {
```

Comment:
Nit: the idiomatic way in go to deal with such errors, that are only scoped to the `if` statement, would be via the short-hand var declaration (so no need for the var declaration above):
```suggestion
	if err := o.K8sGardenClient.Client().Create(ctx, shootState); err != nil {
```

Reply:
I did it because the error declaration is used below:
https://github.com/gardener/gardener/pull/4100/files/805237f0aff31aaf96475d5d3a26bae812f430b5#diff-1b66a0fabd3ab9fcedbaaa435a65b221b2208c2874ce24f07692152e2e32a574R526

Reply:
I need the error variable declaration anyways for https://github.com/gardener/gardener/pull/4100/files/805237f0aff31aaf96475d5d3a26bae812f430b5#diff-1b66a0fabd3ab9fcedbaaa435a65b221b2208c2874ce24f07692152e2e32a574R526